### DISCUSSION TITLE
想实现每10个线程创建一个新连接，报错Alias of 'default' already creating connections

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#18080] [Q&A and General discussion] [answered]
完整报错：
pymilvus.orm.exceptions.ConnectionConfigException: <ConnectionConfigException: (code=0, message=Alias of 'default' already creating connections, but the configure is not the same as passed in.)>
请问应该怎样创建多个连接呢，我哪里写的有问题？
（数据已加载到内存）
import datetime
import random
import time
from pymilvus import connections
from pymilvus import Collection
import threading

def conn(names):
	connections.connect(
		using=names,
		host='xxxx',
		port='19530'
	)
	print("连接成功")
	booK = Collection("nlp9")
	return booK

def Msearch(booK):
	for j in range(3):
		rul = booK.search(
			data=[[1.2121, 1.323232]],
			anns_field="nlp_intro",
			param=search_params,
			limit=3
		)

book = ""
for i in range(100):
	if i % 10 == 0:
		book = conn("twoo"+str(i))
	thr = threading.Thread(target=Msearch, args=(book,))
	thr.start()

**Comment by yhmo** [comment, marked as answer]
```
def conn(conn_name):
    connections.connect(
      alias=conn_name,
      host='xxxx',
      port='19530'
    )
    print("连接成功")
    booK = Collection(name="nlp9", using=conn_name)
    return booK
```
